add example how to use AWSLambdaReceiver (#815)

TheManWhoStaresAtCode opened the pull request:
###  Summary

Provide example how to use AWSLambdaReceiver by adapting the deploy-aws-lambda example accordingly

### Requirements (place an `x` in each `[ ]`)

* [x] I've read and understood the [Contributing Guidelines](https://github.com/slackapi/bolt/blob/main/.github/contributing.md) and have done my best effort to follow them.
* [x] I've read and agree to the [Code of Conduct](https://slackhq.github.io/code-of-conduct).

seratch commented:
@TheManWhoStaresAtCode Thanks for submitting this awesome change!

@mwbrooks I think that we can replace https://github.com/slackapi/bolt-js/tree/%40slack/bolt%403.3.0/examples/deploy-aws-lambda with this rather than adding a new one as `deploy-aws-lambda-receiver`. What do you think?

mwbrooks commented:
@TheManWhoStaresAtCode Awesome, thank you for this pull request!

@seratch I agree, let's replace the current example with this example! It would be better if we can use the original directory path (`/examples/deploy-aws-lambda`) to preserve any online links to that example. But I'm cool with whatever everyone thinks is best!

seratch commented:
>It would be better if we can use the original directory path (/examples/deploy-aws-lambda) to preserve any online links to that example

Yes, I totally agree!

@TheManWhoStaresAtCode Could you update this PR to apply this change to the existing example when you have a chance?

TheManWhoStaresAtCode commented:
As requested I updated the example and also updated the english documentation refering to this example. Unfortunatly, I'm not able to update the japenese documentation.

seratch commented:
The updates on the document look great to me 👍 @mwbrooks Can you have a quick look on the changes?

seratch commented:
@mwbrooks 👋 I will merge this PR soon. Let me know if you have any comments on it.

mwbrooks commented on docs/_deployments/aws-lambda.md:
```diff
@@ -110,48 +110,40 @@ Now that you have an app, let's prepare it for AWS Lambda and the Serverless Fra
-> 💡 This guide requires version `4.x.x` or later.
-
-Next, update the [source code that imports your modules](https://github.com/slackapi/bolt-js-getting-started-app/blob/main/app.js#L1) in `app.js` to require Bolt's Express receiver and the AWS Serverless Express module:
+First, update the [source code that imports your modules](https://github.com/slackapi/bolt-js-getting-started-app/blob/main/app.js#L1) in `app.js` to require Bolt's AwsLambda receiver:
```
Updating the name to be consistent with the usage throughout the doc:
```suggestion
First, update the [source code that imports your modules](https://github.com/slackapi/bolt-js-getting-started-app/blob/main/app.js#L1) in `app.js` to require Bolt's AwsLambdaReceiver:
```

mwbrooks commented on examples/deploy-aws-lambda/README.md:
```diff
@@ -1,10 +1,10 @@
 # Deploying to AWS Lambda ⚡️ Bolt for JavaScript
 
-> Slack app example for deploying to AWS Lambda using the Serverless Framework with Bolt for JavaScript
+> Slack app example for using the AWSLambdaReceiver of Bolt for JavaScript
```
Update the receiver name to be consistent through the doc:
```suggestion
> Slack app example for using the AwsLambdaReceiver of Bolt for JavaScript
```

mwbrooks commented on examples/deploy-aws-lambda/README.md:
```diff
@@ -1,10 +1,10 @@
 ## Overview
 
-This is an example app that updates the [Getting Started ⚡️ Bolt for JavaScript app][bolt-app] to be deployed to [AWS Lambda][aws-lambda] using the [Serverless Framework][serverless-framework].
+This is an example app that updates the [Getting Started ⚡️ Bolt for JavaScript app][bolt-app] to use the AWSLambdaReceiver and be deployed to [AWS Lambda][aws-lambda] using the [Serverless Framework][serverless-framework].
```
Update the receiver name to be consistent through the doc:
```suggestion
This is an example app that updates the [Getting Started ⚡️ Bolt for JavaScript app][bolt-app] to use the AwsLambdaReceiver and be deployed to [AWS Lambda][aws-lambda] using the [Serverless Framework][serverless-framework].
```

mwbrooks commented on examples/deploy-aws-lambda/package.json:
```diff
@@ -1,16 +1,15 @@
+  "name": "bolt-js-aws-lambda-receiver",
   "version": "1.0.0",
-  "description": "Deploying to AWS Lambda ⚡️ Bolt for JavaScript",
+  "description": "Deploying to AWS Lambda ⚡️ Bolt for JavaScript using the AWSLambdaReceiver",
```
Update the receiver name to be consistent through the doc:
```suggestion
  "description": "Deploying to AWS Lambda ⚡️ Bolt for JavaScript using the AwsLambdaReceiver",
```

mwbrooks commented on examples/deploy-aws-lambda/serverless.yml:
```diff
@@ -1,4 +1,4 @@
-service: serverless-bolt-js
+service: serverless-lambda-receiver-bolt-js
```
Update the receiver name to be consistent through the doc:
```suggestion
service: serverless-aws-lambda-receiver-bolt-js
```

mwbrooks reviewed (approved):
Incredible work @TheManWhoStaresAtCode! 🎉 

Sorry about the slow response time to review your PR. It looks fantastic to me! I've added a few "nit" suggestions to keep the `AwsLambdaReceiver` name consistent.

Thank you for taking the time to update this!

TheManWhoStaresAtCode commented:
Thanks @mwbrooks for you positive feedback and providing the tiny improvements. I've updated the PR as suggested.

seratch commented:
LGTM (again) - thanks for improving this!
